src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Comparator;
+
+import seedu.address.logic.commands.SortCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.comparators.PersonFlagComparator;
+import seedu.address.model.person.comparators.PersonNameComparator;
+import seedu.address.model.person.comparators.PersonPrevDateMetComparator;
+
+public class SortCommandParser implements Parser<SortCommand> {
+
+    enum SortArg {
+        MEETING,
+        NAME,
+        PREV,
+        SALARY
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns a Sort Command object for execution.
+     *
+     * @param args arguments to be parsed for SortCommand.
+     * @return SortCommand with parsed Comparator.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public SortCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        String[] sortVariable = trimmedArgs.split("\\s+");
+
+        if (sortVariable.length != 1) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
```

Comment:
Yes, I did consider using the argumentTokenizer. However, that requires the use of prefixes and I felt that for sort it would be easier and more intuitive to just type the parameter to sort by. This is similar to the findCommand which I used as a reference. Thanks for the suggestion!